src/helpers/vcd2lxt.c

```diff
@@ -231,11 +231,8 @@ static unsigned int vcdid_hash(char *s, int len)
  */
 static int vcdsymbsearchcompare(const void *s1, const void *s2)
 {
-    char *v1;
-    struct vcdsymbol *v2;
-
-    v1 = (char *)s1;
-    v2 = *((struct vcdsymbol **)s2);
+    const char *v1 = (const char *)s1;
+    const struct vcdsymbol *v2 = (const struct vcdsymbol *)s2;
```

Comment:
`s2` is a pointer to an element in the sorted symbols array, which contains pointers of type `struct vcdsymbol *`. This means that the original type of `struct vcdsymbol **` for `s2` was correct.